bindings/guile/gnc-optiondb.i

```diff
@@ -1369,7 +1369,11 @@ inline SCM return_scm_value(ValueType value)
                                   GncOptionRangeValue<int>>)
                     {
                         if (scm_is_pair(new_value))
-                            option.set_value(scm_to_int(scm_cdr(new_value)));
+                        {
+                            // in Gnucash 4, the value might have been saved as floating point
+                            auto double_value = scm_to_double(scm_cdr(new_value));
+                            option.set_value(static_cast<int>(double_value));
+                        }
```

Comment:
This will break reading plot size options set by GC5. The SCM input must be tested with `scm_is_integer` and processed accordingly:
```
if (scm_is_pair(new_value))
{
    auto value_num = scm_cdr(new_value);
    if (scm_is_int(value_num))
        option.set_value(scm_to_int(value_num));
    else
        option.set_value(static_cast<int>(scm_to_double(value_num)));
}
```